lib/vector-common/src/finalizer.rs

```diff
@@ -42,9 +42,22 @@ where
     S: FuturesSet<FinalizerFuture<T>> + Default + Send + Unpin + 'static,
 {
     /// Produce a finalizer set along with the output stream of
-    /// received acknowledged batch identifiers.
+    /// received acknowledged batch identifiers. The output stream will end immediately when
+    /// a shutdown signal is received, or when the sender is dropped
     #[must_use]
     pub fn new(shutdown: ShutdownSignal) -> (Self, BoxStream<'static, (BatchStatus, T)>) {
+        FinalizerSet::with_maybe_shutdown(Some(shutdown))
+    }
+
+    /// Produce a finalizer set along with the output stream of
+    /// received acknowledged batch identifiers. The output stream will close after the sender is
+    /// dropped and the stream is drained
+    #[must_use]
+    pub fn new_without_shutdown() -> (Self, BoxStream<'static, (BatchStatus, T)>) {
+        FinalizerSet::with_maybe_shutdown(None)
+    }
+
+    fn with_maybe_shutdown(shutdown: Option<ShutdownSignal>) -> (Self, BoxStream<'static, (BatchStatus, T)>) {
```

Comment:
I think it would be simpler to just make `fn new` take the optional shutdown signal (with the appropriate documentation you have added, of course). I don't see any real benefit to splitting this into separate functions.